modules/govuk_scripts/templates/govuk_node_list_aws.erb

```diff
@@ -55,21 +55,21 @@ def ec2_nodes_list_with_puppet_class(stackname):
                )
 
     reservations = response['Reservations']
-    for reservation in reservations:
-        instances = reservation['Instances']
 
-    return [
-        "%s:%s" % (
-            instance['PrivateDnsName'],
-            next(
-                filter(
-                    lambda t: t['Key'] == 'aws_hostname',
-                    instance['Tags']
-                )
-            )['Value']
-        )
-        for instance in list(instances)
-    ]
+    hostname_and_puppet_class = []
+
+    for reservation in reservations:
+        for instance in list(reservation['Instances']):
+            hostname_and_puppet_class.append("%s:%s" % (
+                instance['PrivateDnsName'],
+                next(
+                    filter(
+                        lambda t: t['Key'] == 'aws_hostname',
+                        instance['Tags']
+                    )
+                )['Value']))
```

Comment:
```python
next(tag['Value'] for tag in instance['Tags'] if tag['Key'] == 'aws_hostname')
```

This would be more pythonic, and potentially easier to read.

Reply:
It did take me a while to get my head around those lines.